tests/context.py

```diff
@@ -54,64 +54,73 @@
 
 try:
 	import sys
-	if sys.__name__ is None:  # pragma: no branch
-		raise ImportError("[CWE-758] OMG! we could not import sys! ABORT. ABORT.")
-except Exception as err:  # pragma: no branch
-	raise ImportError(err)
+	if not hasattr(sys, 'modules') or not sys.modules:  # pragma: no branch
+		raise ModuleNotFoundError("[CWE-440] OMG! sys.modules is not available or empty.") from None
+except ImportError as err:
+	raise ImportError("[CWE-440] Unable to import sys module.") from err
 
 
 try:
 	if 'os' not in sys.modules:
 		import os
 	else:  # pragma: no branch
 		os = sys.modules["""os"""]
-except Exception:  # pragma: no branch
-	raise ImportError("[CWE-440] OS Failed to import.")
+except Exception as err:  # pragma: no branch
+	raise ModuleNotFoundError("[CWE-440] OS Failed to import.") from err
+
+
+try:
+	if 'random' not in sys.modules:
+		import random
+	else:  # pragma: no branch
+		random = sys.modules["""random"""]
+except Exception as err:  # pragma: no branch
+	raise ModuleNotFoundError("[CWE-440] Random Failed to import.") from err
 
 
 try:
 	if 'unittest' not in sys.modules:
 		import unittest
 	else:  # pragma: no branch
 		unittest = sys.modules["""unittest"""]
-except Exception:  # pragma: no branch
-	raise ImportError("[CWE-440] unittest Failed to import.")
+except Exception as err:  # pragma: no branch
+	raise ModuleNotFoundError("[CWE-440] unittest Failed to import.") from err
 
 
 try:
 	if 'Process' not in sys.modules:
 		from multiprocessing import Process as Process  # skipcq: PYL-C0414
 	else:  # pragma: no branch
 		Process = sys.modules["""Process"""]
-except Exception:  # pragma: no branch
-	raise ImportError("[CWE-440] Process Failed to import.")
+except Exception as err:  # pragma: no branch
+	raise ModuleNotFoundError("[CWE-440] Process Failed to import.") from err
 
 
 try:
 	if 'subprocess' not in sys.modules:
 		import subprocess
 	else:  # pragma: no branch
 		subprocess = sys.modules["""subprocess"""]
-except Exception:  # pragma: no branch
-	raise ImportError("[CWE-440] subprocess Failed to import.")
+except Exception as err:  # pragma: no branch
+	raise ModuleNotFoundError("[CWE-440] subprocess Failed to import.") from err
 
 
 try:
 	if 'multicast' not in sys.modules:
 		import multicast  # pylint: disable=cyclic-import - skipcq: PYL-R0401
 	else:  # pragma: no branch
 		multicast = sys.modules["""multicast"""]  # pylint: disable=cyclic-import
-except Exception:  # pragma: no branch
-	raise ImportError("[CWE-440] Python Multicast Repo Failed to import.")
+except Exception as err:  # pragma: no branch
+	raise ImportError("[CWE-440] Python Multicast Repo Failed to import.") from err
```

Comment:
:no_good: Multicast is and exception because it is the module of this project and won't be a "not found" error rather it will be some other unexpected issue with importing, hence the 'ImportError' chained to the caught exception